libraries/rush-lib/src/logic/installManager/WorkspaceInstallManager.ts

```diff
@@ -388,6 +402,22 @@ export class WorkspaceInstallManager extends BaseInstallManager {
     return { shrinkwrapIsUpToDate, shrinkwrapWarnings };
   }
 
+  private _getPackageExtensionChecksum(
```

Comment:
Does an existing method not get exported from the pnpm packages? If we could re-use their code, I think that would be ideal. If not, then this looks good :)

Reply:
`createObjectChecksum` is exported by `@pnpm/core`. However if we add that as dependency of Rush, it brings in a whole lots of other unused dependencies.

Reply:
Would be a good idea to add a comment that permalinks to the implementation in PNPM source then, just for future reference (should anything ever change... hopefully unlikely, but good to have that reference)

Reply:
Added